contracts/modules/Checkpoint/ERC20DividendCheckpoint.sol

```diff
@@ -1,157 +1,205 @@
+     * @param _maturity Time from which dividend can be paid
+     * @param _expiry Time until dividend can no longer be paid, and can be reclaimed by issuer
+     * @param _token Address of ERC20 token in which dividend is to be denominated
+     * @param _amount Amount of specified token for dividend
+     * @param _excluded List of addresses to exclude
+     * @param _name name/title for identification
+     */
+    function createDividendWithExclusions(uint256 _maturity, uint256 _expiry, address _token, uint256 _amount, address[] _excluded, string _name) public onlyOwner {
+        uint256 checkpointId = ISecurityToken(securityToken).createCheckpoint();
+        _createDividendWithCheckpointAndExclusions(_maturity, _expiry, _token, _amount, checkpointId, _excluded, _name);
+    }
+
+    /**
+     * @notice Creates a dividend with a provided checkpoint
+     * @param _maturity Time from which dividend can be paid
+     * @param _expiry Time until dividend can no longer be paid, and can be reclaimed by issuer
+     * @param _token Address of ERC20 token in which dividend is to be denominated
+     * @param _amount Amount of specified token for dividend
+     * @param _checkpointId Checkpoint id from which to create dividends
+     * @param _excluded List of addresses to exclude
+     * @param _name name/title for identification
+     */
+    function createDividendWithCheckpointAndExclusions(
+        uint256 _maturity, 
+        uint256 _expiry, 
+        address _token, 
+        uint256 _amount, 
+        uint256 _checkpointId, 
+        address[] _excluded,
+        string _name
+    ) 
+        public
+        onlyOwner      
+    {
+        _createDividendWithCheckpointAndExclusions(_maturity, _expiry, _token, _amount, _checkpointId, _excluded, _name);
+    }
+
+    /**
+     * @notice Creates a dividend with a provided checkpoint
+     * @param _maturity Time from which dividend can be paid
+     * @param _expiry Time until dividend can no longer be paid, and can be reclaimed by issuer
+     * @param _token Address of ERC20 token in which dividend is to be denominated
+     * @param _amount Amount of specified token for dividend
+     * @param _checkpointId Checkpoint id from which to create dividends
+     * @param _excluded List of addresses to exclude
+     * @param _name name/title for identification
+     */
+    function _createDividendWithCheckpointAndExclusions(
+        uint256 _maturity, 
+        uint256 _expiry, 
+        address _token, 
+        uint256 _amount, 
+        uint256 _checkpointId, 
+        address[] _excluded,
+        string _name
+    ) 
+        internal  
+    {
+        require(_excluded.length <= EXCLUDED_ADDRESS_LIMIT, "Too many addresses excluded");
+        require(_expiry > _maturity, "Expiry is before maturity");
+        require(_expiry > now, "Expiry is in the past");
+        require(_amount > 0, "No dividend sent");
+        require(_token != address(0), "0x not valid token");

```

Comment:
add the sanity check on the `_name`. `require(bytes(_name).length > 0);`

Reply:
Changes done